blocks/image-link-slider/image-link-slider.js

```diff
@@ -0,0 +1,101 @@
+                            </a>
+                        </div>
+                    </div>
+                    </div>
+                `;
+    carouselItems.append(slide);
+  });
+}
+
+async function loadCarousel(block, carouselItems) {
+  const carouselBlock = buildBlock('carousel', '');
+  carouselBlock.innerHTML = '';
+  block.classList.forEach((className) => {
+    carouselBlock.classList.add(className);
+  });
+
+  Array.from(carouselItems.children).forEach((carouselItemElement) => {
+    const divElement = document.createElement('div');
+    Array.from(carouselItemElement.children).forEach((child) => {
+      divElement.appendChild(child.cloneNode(true));
+    });
+    carouselBlock.appendChild(divElement);
+  });
+
+  const carouselBlockParent = document.createElement('div');
+  carouselBlockParent.classList.add('carousel-wrapper');
+  carouselBlockParent.appendChild(carouselBlock);
+  block.parentElement.insertAdjacentElement('afterend', carouselBlockParent);
+  decorateBlock(carouselBlock);
+
+  carouselBlock.querySelectorAll('.social-share').forEach((anchor) => anchor.addEventListener('click', () => handleSocialShareClick(anchor)));
+  return loadBlock(carouselBlock);
+}
+
+function handleTitleConfig(titleElement, container) {
+  const titleText = titleElement.textContent.trim();
+  const title = document.createElement('h2');
+  title.textContent = titleText;
+
+  const titleWrapper = document.createElement('div');
+  titleWrapper.classList.add('title-wrapper');
+  titleWrapper.appendChild(title);
+
+  container.insertBefore(titleWrapper, container.firstChild);
+}
+
+export default async function decorate(block) {
+  block.style.display = 'none';
+  const container = block.closest('.section .section-container');
+  container.style.display = 'none';
+  const configElementsArray = Array.from(block.children);
+
+  const promises = configElementsArray.map(async (configElement) => {
+    const configNameElement = configElement.querySelector('div');
+    const configName = configNameElement.textContent.trim().toLowerCase();
+    if (configName === 'api') {
+      const carouselItems = document.createElement('div');
+      carouselItems.classList.add('carousel-items');
+      const apiName = configNameElement.nextElementSibling.textContent.trim();
+      const data = await callAPI(apiName);
+      if (block.classList.contains('image-link-slider')) {
+        renderImageLinkVariant(data, carouselItems);
+      }
+      await loadCarousel(block, carouselItems);
+    } else if (configName === 'title') {
+      const titleElement = configNameElement.nextElementSibling;
+      handleTitleConfig(titleElement, container);
+    } else if (configName === 'link') {
+      const buttonWrapper = document.createElement('div');
+      buttonWrapper.classList.add('button-wrapper');
+      buttonWrapper.append(configNameElement.nextElementSibling);
+      container.append(buttonWrapper);
+    }
+  });
+
+  await Promise.all(promises);
```

Comment:
Similar as prior comment, I think we should avoid awaiting as much as possible and use placeholders and fill up content as it arrives from the server asynchronously.

Reply:
> Similar as prior comment, I think we should avoid awaiting as much as possible and use placeholders and fill up content as it arrives from the server asynchronously.

done.